DiabloII.Application.Tests/Steps/Suggestions/GetAllHalLinks/GetAllHalLinksSuggestionsSteps.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using DiabloII.Application.Tests.Apis.Domains.Suggestions;
+using DiabloII.Application.Tests.Contexts.Domains.Suggestions;
+using Shouldly;
+using TechTalk.SpecFlow;
+
+namespace DiabloII.Application.Tests
+{
+    [Binding]
+    [Scope(Tag = "suggestions")]
+    public class GetAllHalLinksSuggestionsSteps
+    {
+        private readonly ISuggestionsApi _suggestionsApi;
+      
+        private readonly IHalSuggestionsTestContext _halSuggestionsContext;
+
+        public GetAllHalLinksSuggestionsSteps(ISuggestionsApi suggestionsApi, IHalSuggestionsTestContext halSuggestionsContext)
+        {
+            _suggestionsApi = suggestionsApi;
+            _halSuggestionsContext = halSuggestionsContext;
+        }
+
+        [When(@"I get all suggestions with hal links")]
+        public async Task WhenIGetAllSuggestionsWithHalLinks() =>
+            _halSuggestionsContext.HalResources = await _suggestionsApi.GetAllWithHals();
+
+        [Then(@"I should retrieve ""(.*)"" as suggestions hal links")]
+        public void ThenIShouldRetrieveAsSuggestionsHalLinks(string halLinks)
+        {
+            var halResources = _halSuggestionsContext.HalResources;
+            var actualSuggestionHalLinks = halResources._Links.Keys;
+            var actualVoteHalLinks = halResources.Elements.SelectMany(e => e.Votes).SelectMany(vote => vote._Links.Keys);
+            var actualCommentHalLinks = halResources.Elements.SelectMany(e => e.Comments).SelectMany(comment => comment._Links.Keys);
```

Comment:
That's not the most performant algoritm to determine if links are presents because of the SelectMany & Union.

In order to optimize this, I should fill a list and use simple foreachs.

But the volumetry tested don't worth it at all so I prefer to get an explicit and readable code.